Src/libCZI/utilities.h

```diff
@@ -204,3 +204,74 @@ struct ParseEnumHelper
     }
 };
 
+/// This class allows to calculate the area covered by a set of rectangles. The mode of operation
+/// is:
+/// - Create an instance of the class and add the rectangles to it (using AddRectangle) .
+/// - Then, for a given rectangle, call CalcAreaOfIntersectionWithRectangle in order to get the area of the intersection  
+///    of this rectangle with the union of the rectangles added before.
+/// The rectangles being added do not have to follow any order, or are required to be non-overlapping.
+class RectangleCoverageCalculator
+{
+private:
+    /// This vector contains the rectangles added to the state of the instance. The rectangles
+    /// in this vector are guaranteed to be non-overlapping. That is why the name 'splitters' is used -
+    /// if rectangles are which are overlapping with the existing ones, the rectangle is split smaller 
+    /// rectangles (which we call the 'splitters') which are non-overlapping with the existing ones.
+    std::vector<libCZI::IntRect> splitters_;    
+public:
+    /// Adds a rectangle to the state.
+    ///
+    /// \param  rectangle   The rectangle to be added.
+    void AddRectangle(const libCZI::IntRect& rectangle);
+
+    /// Adds the rectangles given by the iterator to the state of the instance.
+    ///
+    /// \typeparam  tIterator   Type of the iterator.
+    /// \param  begin   The begin.
+    /// \param  end     The end.
+    template <typename tIterator>
+    void AddRectangles(tIterator begin, tIterator end)
+    {
+        for (tIterator it = begin; it != end; ++it) 
+        {
+            this->AddRectangle(*it);
+        }
+    }
+
+    /// Calculates the area of intersection of the specified rectangle with the
+    /// union of the rectangles added before.
+    /// If the query_rectangle rectangle is invalid, the return value is 0.
+    ///
+    /// \param  query_rectangle   The query rectangle.
+    ///
+    /// \returns    The calculated area of intersection of the specified rectangle with the
+    ///             union of the rectangles added before.
+    std::int64_t CalcAreaOfIntersectionWithRectangle(const libCZI::IntRect& query_rectangle) const;
+
+    /// Query if 'rectQuery' is completely covered is completely covered by the union of the rectangles added before.
+    /// If the query_rectangle rectangle is invalid, the return value is true.
```

Comment:
```suggestion
    /// Returns a value indicating whether 'rectQuery' is completely covered by the union of the rectangles added before.
    /// If the query_rectangle rectangle is invalid, the return value is true.
```

Reply:
And we should maybe indicate that this is an O(n_splitters) operation.

Reply:
done